chall07/migferna.go

```diff
@@ -0,0 +1,47 @@
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"strconv"
+	"strings"
+)
+
+func rgbToHexadecimal(response string) (int, int, int, int) {
+	str := strings.Split(response, " ")
+	data := str[0]
+	values := strings.Split(data, ",")
+	id, error := strconv.Atoi(strings.Split(values[0], "=")[1])
+	red, error := strconv.Atoi(strings.Split(values[1], "=")[1])
+	green, error := strconv.Atoi(strings.Split(values[2], "=")[1])
+	blue, error := strconv.Atoi(strings.Split(values[3], "=")[1])
+
+	if error != nil {
+		log.Fatalln(error)
+	}
+
+	return id, red, green, blue
+}
+
+func makeRequest(url string) string {
+	response, error := http.Get(url)
+	body, error := ioutil.ReadAll(response.Body)
+
+	if error != nil {
+		log.Fatalln(error)
+	}
+
+	defer response.Body.Close()
```

Comment:
The goal of defer is to close the reader, NO MATTER WHAT HAPPENS. Here if the `ioutil.ReadAll`  fails for some reason you won't be closing the body.
Also, you are not handling the error sent by `http.Get`, which means you will potentially try to access `response.Body` whilst `response` being `nil`. This will cause an unexpected `panic`.

You should handle the error sent by `http.Get` and make a defer right after that:
```suggestion
	response, error := http.Get(url)
	if error != nil {
		log.Fatalln(error)
	}
	defer response.Body.Close()
	
	body, error := ioutil.ReadAll(response.Body)
	if error != nil {
		log.Fatalln(error)
	}
```